contracts/extensions/ExpenditureUtils.sol

```diff
@@ -0,0 +1,222 @@
+
+  // Storage
+
+  uint256 stakeFraction;
+
+  mapping (address => mapping (uint256 => uint256)) stakes;
+
+  // Modifiers
+
+  modifier onlyRoot() {
+    require(colony.hasUserRole(msgSender(), 1, ColonyDataTypes.ColonyRole.Root), "evaluated-expenditure-caller-not-root");
+    _;
+  }
+
+  // Overrides
+
+  /// @notice Returns the identifier of the extension
+  function identifier() public override pure returns (bytes32) {
+    return keccak256("ExpenditureUtils");
+  }
+
+  /// @notice Returns the version of the extension
+  function version() public override pure returns (uint256) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    require(address(colony) == address(0x0), "extension-already-installed");
+
+    colony = IColony(_colony);
+  }
+
+  /// @notice Called when upgrading the extension
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension
+  function deprecate(bool _deprecated) public override auth {
+    deprecated = _deprecated;
+  }
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
+    selfdestruct(address(uint160(address(colony))));
+  }
+
+  // Public
+
+  function setStakeFraction(uint256 _stakeFraction) public onlyRoot {
+    stakeFraction = _stakeFraction;
+  }
+
+  function makeExpenditureWithStake(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _domainId,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    uint256 domainRep = getReputationFromProof(_domainId, _key, _value, _branchMask, _siblings);
+    uint256 stake = wmul(domainRep, stakeFraction);
+
+    colony.obligateStake(msgSender(), _domainId, stake);
+    colony.makeExpenditure(_permissionDomainId, _childSkillIndex, _domainId);
+
+    uint256 expenditureId = colony.getExpenditureCount();
+    stakes[msgSender()][expenditureId] = stake;
+    colony.transferExpenditure(expenditureId, msgSender());
+  }
+
+  function slashStake(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _expenditureId
+  )
+    public
+  {
+    ColonyDataTypes.Expenditure memory expenditure = colony.getExpenditure(_expenditureId);
+    uint256 stake = stakes[expenditure.owner][_expenditureId];
+    require(stake > 0, "expenditure-utils-nothing-to-claim");
+
+    require(
+      colony.hasInheritedUserRole(msgSender(), _permissionDomainId, ColonyDataTypes.ColonyRole.Arbitration, _childSkillIndex, expenditure.domainId),
+      "expenditure-utils-caller-not-arbitration"
+    );
+
+    delete stakes[expenditure.owner][_expenditureId];
+    colony.transferStake(_permissionDomainId, _childSkillIndex, address(this), expenditure.owner, expenditure.domainId, stake, address(0x0));
```

Comment:
Wasn't sure where to put them... we could send them to the colony, but there isn't really functionality to withdraw them from tokenLocking. Perhaps that's something that should be added.

Reply:
If the spec is ambiguous, then we should ask product :smile:

Reply:
Good point. How does the staking work for motions? I was under the impression the activated tokens were locked in the Motions & Disputes Extension contract. Having to replicate that functionality in Expenditures does seem repetitive, especially if this staking functionality is used by another extension in future.

My thoughts would be to send them to the Colony and then add withdraw functionality. Then it could be used by other extensions in future.

Reply:
The functionality isn't duplicated -- here as well as with motions the stakes are managed by the token locking contract. The difference is that with motions, all the stakes are ultimately returned to other users. Here, there's no single user whom it would make sense to send tokens to. We could return them to the colony, but that would involve adding functionality allowing a colony to withdraw tokens from token locking (which is not hard to do nor necessarily a bad idea, it just doesn't exist yet). One thing to consider though is how returning stakes to the colony (as opposed to burning them) could create perverse incentives to slash people in order to seize their funds. Burning the tokens avoids this issue.

Reply:
>One thing to consider though is how returning stakes to the colony (as opposed to burning them) could create perverse incentives to slash people in order to seize their funds

You're right, but it feels pretty unlikely to me. A colony behaving in this way is surely shooting themselves in the foot (long-term)?

Either way, these tokens haven't been burned. They're under the control of the TokenLocking contract (which claims only `0x0` can access them, but Token Locking can be upgraded). If we want to burn them, I'd say need to be put under the control of `0x0` (i.e. added to the balance of that address, according to the token contract). Actually burning them, if possible, would be nice though.

Reply:
After confirming with @jakzilla, it looks like the preference is to go with burning.